torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -722,11 +905,54 @@ bool extentCompare(
 } // namespace
 
 bool HaloInfo::extentLessEqual(IterDomain* id1, IterDomain* id2) const {
-  return extentCompare(*this, id1, id2, std::less_equal<unsigned>());
+  auto cmp = [](kir::Int* x, kir::Int* y) {
+    if (x == y) {
+      return true;
+    }
+    auto xv = x->value();
+    auto yv = y->value();
+    return xv.has_value() && yv.has_value() && xv.value() <= yv.value();
+  };
+  return extentCompare(*this, id1, id2, cmp);
 }
 
 bool HaloInfo::extentEqual(IterDomain* id1, IterDomain* id2) const {
-  return extentCompare(*this, id1, id2, std::equal_to<unsigned>());
+  std::function<bool(kir::Int*, kir::Int*)> cmp = [&](kir::Int* x,
```

Comment:
Could you add some comments on assumptions in this comparison? It seems it can't catch all cases, but if both DAGs producing the ints are exactly the same including all inputs then it will detect it is the same.

Reply:
Yes, it just checks if two `kir::Int` objects have the same expression trees. As stated in the comment in the header, this is a conservative analysis and only returns true when the equality is proven. That is just sufficient for the current use case, but it could be refactored to something more generic if necessary. Added comments.